delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/model/SparqlLink.scala

```diff
@@ -0,0 +1,144 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.model
+
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.client.SparqlResults.Binding
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary.nxv
+import io.circe.syntax.EncoderOps
+import io.circe.{Encoder, Json, JsonObject}
+
+import java.time.Instant
+import scala.util.Try
+
+sealed trait SparqlLink {
+
+  /**
+    * @return the @id value of the resource
+    */
+  def id: Iri
+
+  /**
+    * @return the collection of types of this resource
+    */
+  def types: Set[Iri]
+
+  /**
+    * @return the paths from where the link has been found
+    */
+  def paths: List[Iri]
+}
+
+object SparqlLink {
+
+  /**
+    * A link that represents a managed resource on the platform.
+    *
+    * @param id            the @id value of the resource
+    * @param project       the @id of the project where the resource belongs
+    * @param self          the access url for this resources
+    * @param rev           the revision of the resource
+    * @param types         the collection of types of this resource
+    * @param deprecated    whether the resource is deprecated of not
+    * @param created       the instant when this resource was created
+    * @param updated       the last instant when this resource was updated
+    * @param createdBy     the identity that created this resource
+    * @param updatedBy     the last identity that updated this resource
+    * @param constrainedBy the schema that this resource conforms to
+    * @param paths         the paths from where the link has been found
+    */
+  final case class SparqlResourceLink(
```

Comment:
All of this is very suspiciously similar to `ResourceF`. Couldn't we reuse it and have a `ResourceF[A]` where A in your case is `paths` ?

Reply:
We have two cases: 
`SparqlResourceLink` and`SparqlExternalLink`. `SparqlResourceLink` is almost ResourceF and probably we could use it for that, but SparqlExternalLink has only `id`, `paths` and `types`  and in this case ResourceF contains a lot of fields which are not available in `SparqlExternalLink`.

Reply:
Yeah. So the case `SparqlResourceLink` should wrap a resourceF

Reply:
What would be the value of `rev`, `deprecated`, `createdAt`, `updatedAt`, `createdBy`, `updatedBy` and `schema` for a `ResourceF` which would represent what we currently have as `SparqlExternalLink`?